model/DataObject.php

```diff
@@ -1902,13 +1898,13 @@ public function belongsToComponent($component, $classOnly = true) {
 	}
 
 	/**
-	 * Return all of the database fields defined in self::$db and all the parent classes.
-	 * Doesn't include any fields specified by self::$has_one.  Use $this->hasOne() to get these fields
+	 * Return all of the database fields in this object
 	 *
 	 * @param string $fieldName Limit the output to a specific field name
+	 * @param string $table Out parameter of the table this db field is set to
 	 * @return array The database fields
 	 */
-	public function db($fieldName = null) {
+	public function db($fieldName = null, &$table = null) {
```

Comment:
I could change the return type to be `Table.Column(spec)` format instead of `Column(spec)`. :)